### PR TITLE
worker job: Make hard-coded list of files to upload configurable

### PR DESCRIPTION
**Pull request by asmorodskyi**
Replace hard-coded list to upload with variable

This replaces hard-coded list of files which uploaded from pools
dir with variable which will be defined in job.

### CODE REVIEW

File: lib/OpenQA/Worker/Job.pm [hunk 1]
```diff
@@ -439,9 +439,12 @@ sub _stop_step_5_upload {
         }
 
         # upload other logs and files
-        for my $file (
-            qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt)
-          )
+        my @files_to_upload = qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt);
+        if (defined $self->info->{settings}->{VIRTIO_CONSOLE_NUM} && $self->info->{settings}->{VIRTIO_CONSOLE_NUM} > 1) {
+            # in case second virtio_console defined we want to get its log too
+            push(@files_to_upload, 'virtio_console1.log');
```
Comment (cfconrad) [inline lib/OpenQA/Worker/Job.pm]: I avoided to upload `virio_consoleX.log` for reason. As I used it for big amount of data. Maybe we should enable it explicit? WDYT?

If we upload depending on VIRTIO_CONSOLE_NUM I would upload all `virtio_consoleX.log` files if present.

Comment (okurz) [reply lib/OpenQA/Worker/Job.pm]: this code is even very much os-autoinst centric. Maybe it would be better to supply that list from os-autoinst directly, e.g. in a variable, WDYT?

File: lib/OpenQA/Worker/Job.pm [hunk 2]
```diff
@@ -913,8 +916,8 @@ sub post_upload_progress_to_liveviewhandler {
 
     my $job_id = $self->id;
     $self->client->send(
-        post => "/liveviewhandler/api/v1/jobs/$job_id/upload_progress",
-        service_port_delta => 2,                     # liveviewhandler is supposed to run on web UI port + 2
+        post               => "/liveviewhandler/api/v1/jobs/$job_id/upload_progress",
+        service_port_delta => 2,                                                        # liveviewhandler is supposed to run on web UI port + 2
```
Comment (okurz) [inline lib/OpenQA/Worker/Job.pm]: the additional indention on the comment looks broken, please try to put it above the line

Comment (asmorodskyi) [reply lib/OpenQA/Worker/Job.pm]: I just reverted everything. this was mistake to include unrelated lines formatting   into this PR

**Comment by okurz** [comment]
@asmorodskyi the commit message as well, don't try to fool me ;)

**Comment by asmorodskyi** [comment]
>this code is even very much os-autoinst centric. Maybe it would be better to supply that list from >os-autoinst directly, e.g. in a variable, WDYT?


@okurz something like this + patch to os-autoinst ?

File: lib/OpenQA/Worker/Job.pm [hunk 3]
```diff
@@ -439,23 +439,27 @@ sub _stop_step_5_upload {
...
-            # replace some file names
-            my $ofile = $file;
-            $ofile =~ s/serial0/serial0.txt/;
-            $ofile =~ s/virtio_console.log/serial_terminal.txt/;
-
-            my %upload_parameter = (file => {file => "$pooldir/$file", filename => $ofile},);
-            if (!$self->_upload_log_file_or_asset(\%upload_parameter)) {
-                $reason = 'api-failure';
-                last;
+        if (defined $self->info->{settings}->{AUOTINST_UPLOAD_FROM_POOL}) {
```
Comment (okurz) [inline lib/OpenQA/Worker/Job.pm]: Without the typo please :)

File: lib/OpenQA/Worker/Job.pm [hunk 4]
```diff
@@ -439,23 +439,27 @@ sub _stop_step_5_upload {
...
+
+                my %upload_parameter = (file => {file => "$pooldir/$file", filename => $ofile},);
+                if (!$self->_upload_log_file_or_asset(\%upload_parameter)) {
+                    $reason = 'api-failure';
+                    last;
+                }
             }
         }
+        else {
+            log_error('AUOTINST_UPLOAD_FROM_POOL is not defined nothing will be uploaded from pool directory');
```
Comment (okurz) [inline lib/OpenQA/Worker/Job.pm]: Better put the old list in as default. Instead of if/else use `//`

File: lib/OpenQA/Worker/Job.pm (hunk 3 above)
Comment (asmorodskyi) [reply lib/OpenQA/Worker/Job.pm]: Done

File: lib/OpenQA/Worker/Job.pm (hunk 4 above)
Comment (asmorodskyi) [reply lib/OpenQA/Worker/Job.pm]: yeah agree with idea to fallback to old list , but let's do something in the middle . WDYT ? Still find log message needed

**Comment by asmorodskyi** [comment]
@okurz so there is no need for second patch in os-autoinst ? we will just add this variable only to jobs where we want to add extra files for upload ? Maybe we need to mention this variable in documentation ? which one exactly ?

File: lib/OpenQA/Worker/Job.pm (hunk 4 above)
Comment (okurz) [reply lib/OpenQA/Worker/Job.pm]: and error in the system log of the worker? Who would look at this message?

File: lib/OpenQA/Worker/Job.pm [hunk 5]
```diff
@@ -439,9 +439,15 @@ sub _stop_step_5_upload {
...
-            qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt)
-          )
+        my @files_to_upload;
+        if (defined $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL}) {
+            @files_to_upload = split(',', $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL});
+        }
+        else {
+            log_error('AUTOINST_UPLOAD_FROM_POOL is not defined fallback to default list');
+            @files_to_upload = qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt);
+        }
```
Comment (okurz) [inline lib/OpenQA/Worker/Job.pm]: Replace the whole block by

```perl
use constant UPLOAD => qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt);
…
my @files_to_upload = split(',', $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL} // '') // UPLOAD;
```

File: lib/OpenQA/Worker/Job.pm [hunk 6]
```diff
@@ -457,6 +463,7 @@ sub _stop_step_5_upload {
             }
         }
 
+
```
Comment (okurz) [inline lib/OpenQA/Worker/Job.pm]: unrelated whitespace change, please revert

File: lib/OpenQA/Worker/Job.pm (hunk 5 above)
Comment (cfconrad) [reply lib/OpenQA/Worker/Job.pm]: I'm not sure if it produce the output what you expect. Look at this:
```
perl -e "use Data::Dumper; @a = qw/foo bla/; @b = split(',','') // @a; print Dumper(\@b);"
```

Comment (okurz) [reply lib/OpenQA/Worker/Job.pm]: just use `||` then

File: lib/OpenQA/Worker/Job.pm [hunk 7]
```diff
@@ -439,9 +441,8 @@ sub _stop_step_5_upload {
         }
 
         # upload other logs and files
-        for my $file (
-            qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt)
-          )
+        my @files_to_upload = split(',', $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL} // '') // UPLOAD_FROM_POOL;
```
Comment (cfconrad) [inline lib/OpenQA/Worker/Job.pm]: Suggest:
```suggestion
        my @files_to_upload = split(',', $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL} // '') || UPLOAD_FROM_POOL;
```

Comment (cfconrad) [reply lib/OpenQA/Worker/Job.pm]: Another thing which is "os-autoinst" related are these rename things like: https://github.com/os-autoinst/openQA/pull/2269/files#diff-2559347dfdfd46e3d678b4f8823f2d7fR452
Do we need to make this somehow also configurable?

Comment (okurz) [reply lib/OpenQA/Worker/Job.pm]: I don't know why the replacement is done here and not already done correctly in os-autoinst. I don't see a compelling reason to not propose a change in os-autoinst. But making it configurable might be beneficial as well.

Comment (asmorodskyi) [reply lib/OpenQA/Worker/Job.pm]: > Another thing which is "os-autoinst" related are these rename things like: https://github.com/os-autoinst/openQA/pull/2269/files#diff-2559347dfdfd46e3d678b4f8823f2d7fR452
> Do we need to make this somehow also configurable?


I personally don't see sense in such replacement , so think this line should be deleted . But I think it worth a separate PR and separate discussion as there might be people who will be against "renaming" of serial_terminal.txt . Or maybe actual fix for line which you mention is rename virtio_console.log into serial_terminal.log ?

Comment (okurz) [reply lib/OpenQA/Worker/Job.pm]: Providing files as .txt will probably provide a hint to browsers that the mime type is different so that the browser knows, based on user preferences, how to handle simple text files, e.g. show them directly in the browser instead of downloading as for binary files. However that part never worked properly for me, either I am always offered a download or the browser tries to open binary files directly in browser with garbled text.

**Comment by okurz** [comment]
@asmorodskyi I hope I am not annoying but I could see in the commit you just added again an example of describing in the commit message what you changed without describing the reasoning properly. I would appreciate if you can change that, should be easy, e.g. "Make … configurable" and everybody should understand: "ah, so now I can customize, cool" :)

File: lib/OpenQA/Worker/Job.pm (hunk 7 above)
Comment (cfconrad) [reply lib/OpenQA/Worker/Job.pm]: I think the reason for the replacement is, that a "common" user get the same output names, unrelated on specific background implementations.

To make it configurable, I would just allow a syntax like `realname=>destname` in `AUTOINST_UPLOAD_FROM_POOL` and use it if present.

Comment (asmorodskyi) [reply lib/OpenQA/Worker/Job.pm]: > Providing files as .txt will probably provide a hint to browsers that the mime type is different so that the browser knows, based on user preferences, how to handle simple text files, e.g. show them directly in the browser instead of downloading as for binary files. However that part never worked properly for me, either I am always offered a download or the browser tries to open binary files directly in browser with garbled text.

I am not against ".txt" ending of logs I am against to have them different in pool directory and as resulting asset. Do you see any issue to name  `virtio_console.log` as `serial_terminal.txt` initially to not bother with renaming at all ?

Comment (cfconrad) [reply lib/OpenQA/Worker/Job.pm]: no, looks also fine for me.

**Comment by cfconrad** [comment]
@asmorodskyi plz add this variable to https://github.com/os-autoinst/os-autoinst/blob/master/doc/backend_vars.asciidoc . @okurz  or is there a better place for variables in openQA?

File: lib/OpenQA/Worker/Job.pm [hunk 8]
```diff
@@ -439,9 +441,8 @@ sub _stop_step_5_upload {
         }
 
         # upload other logs and files
-        for my $file (
-            qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt)
-          )
+        my @files_to_upload = split(',', $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL} // '') || UPLOAD_FROM_POOL;
```
Comment (perlpunk) [inline lib/OpenQA/Worker/Job.pm]: This is not workingas intended:
```perl
my $settings = ""; 
use constant FOO => qw/ a b c d /;
my @x = (split m/,/, $settings // "") || FOO;
say "x=(@x)"
# Output: x=(a b c d)

my $settings = "a,b";
use constant FOO => qw/ a b c d /;
my @x = (split m/,/, $settings // "") || FOO;
say "x=(@x)"
# Output: x=(2)
```

Comment (perlpunk) [reply lib/OpenQA/Worker/Job.pm]: I suggest:
```perl
my @files_to_upload = split m/,/, $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL} // '';
@files_to_upload = UPLOAD_FROM_POOL unless @files_to_upload;
```

Comment (asmorodskyi) [reply lib/OpenQA/Worker/Job.pm]: I just tried to run following code - 
```
my @files_to_upload = split m/,/, $settings;
@files_to_upload = qw(w e r) unless @files_to_upload;
print "x=(@files_to_upload)";
# Output : x=(w e r) 
# no warnings about use of uninitialized variable ( looks like split handling this somehow ... ) 
```

so I will use your version but without `// ''` or I missing something here ?

File: lib/OpenQA/Worker/Job.pm [hunk 9]
```diff
@@ -439,9 +441,9 @@ sub _stop_step_5_upload {
         }
 
         # upload other logs and files
-        for my $file (
-            qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt)
-          )
+        my @files_to_upload = split(m/,/, $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL});
+        @files_to_upload = UPLOAD_FROM_POOL unless @files_to_upload;
```
Comment (cfconrad) [inline lib/OpenQA/Worker/Job.pm]: This would also not allow you to disable all files to upload with a setting link `AUTOINST_UPLOAD_FROM_POOL=''`

Comment (asmorodskyi) [reply lib/OpenQA/Worker/Job.pm]: didn't get your point - you think we should allow this ?

Comment (cfconrad) [reply lib/OpenQA/Worker/Job.pm]: Yes, for me it is consistence. If I set `AUTOINST_UPLOAD_FROM_POOL` to what ever, what ever is used.

Comment (asmorodskyi) [reply lib/OpenQA/Worker/Job.pm]: > Yes, for me it is consistence. If I set `AUTOINST_UPLOAD_FROM_POOL` to what ever, what ever is used.

are you sure ? 
Just test it with code : 
```
my $settings = ' ';
my @files_to_upload = split m/,/, $settings;
@files_to_upload = qw(w e r) unless @files_to_upload;
print "x=(@files_to_upload)";
# Output : x=( ) 
```

**Comment by asmorodskyi** [comment]
so @okurz  , @perlpunk  what do you think about this version ?

**Comment by asmorodskyi** [comment]
aaand one more version suggested by @cfconrad , I think it is actually best one . WDYT ?

File: lib/OpenQA/Worker/Job.pm [hunk 10]
```diff
@@ -439,9 +441,8 @@ sub _stop_step_5_upload {
         }
 
         # upload other logs and files
-        for my $file (
-            qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt)
-          )
+        my @files_to_upload = split(m/,/, $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL} // UPLOAD_FROM_POOL) ;
```
Comment (okurz) [inline lib/OpenQA/Worker/Job.pm]: delete the space before the `;`

**Comment by asmorodskyi** [comment]
> ok, if it works. I suggest for variables used within the context of openQA to also be documented within openQA documents, not os-autoinst, at least not as primary source of information. You can still mention it in os-autoinst though.
> 
> So please make sure that this PR also includes the relevant documentation changes.

sorry but I fail to understand which file in **this** project would be proper place for this variable

File: lib/OpenQA/Worker/Job.pm (hunk 8 above)
Comment (perlpunk) [reply lib/OpenQA/Worker/Job.pm]: did you miss a `use warnings`?

```
perl -wE'
my $settings;
my @x = split m/,/, $settings;' 
Use of uninitialized value $settings in split at -e line 3.
```

I always run my oneliners with `-w`

Comment (asmorodskyi) [reply lib/OpenQA/Worker/Job.pm]: yeah looks like , my bad . But anyway this version is dropped already

**Comment by okurz** [comment]
Best place I could find for documentation is in https://github.com/os-autoinst/openQA/blob/master/docs/WritingTests.asciidoc#using-text-consoles-and-the-serial-terminal because the section also mentions the explicit file names, e.g. serial0.txt

File: lib/OpenQA/Worker/Job.pm [hunk 11]
```diff
@@ -439,9 +441,8 @@ sub _stop_step_5_upload {
         }
 
         # upload other logs and files
-        for my $file (
-            qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt)
-          )
+        my @files_to_upload = split(m/,/, $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL} // UPLOAD_FROM_POOL);
```
Comment (Martchus) [inline lib/OpenQA/Worker/Job.pm]: It would be better to use `$self->settings->{AUTOINST_UPLOAD_FROM_POOL}` here instead. Generally jobs without settings are accepted and in this case `$self->info->{settings}` would be `undef`. `$self->settings` is always supposed to return a HASH-ref.

Comment (asmorodskyi) [reply lib/OpenQA/Worker/Job.pm]: thanks for the hint ! done

File: docs/WritingTests.asciidoc [hunk 12]
```diff
@@ -1408,3 +1408,13 @@ state of the specified git refspec. For example the asset caching
 functionality will override the +PRODUCTDIR+ variable. The above method can
 still be used in this case in the common case that the test schedule does not
 need to be changed or in case a custom schedule is defined by +SCHEDULE+.
+
+==Controling list of log files to upload
+
+During every job run worker produces various log files.
+By specifying +AUTOINST_UPLOAD_FROM_POOL+ variable with comma separated list of variables
```
Comment (cfconrad) [inline docs/WritingTests.asciidoc]:
```suggestion
By specifying +AUTOINST_UPLOAD_FROM_POOL+ variable with comma separated list of filenames
```

**Comment by asmorodskyi** [comment]
> Best place I could find for documentation is in https://github.com/os-autoinst/openQA/blob/master/docs/WritingTests.asciidoc#using-text-consoles-and-the-serial-terminal because the section also mentions the explicit file names, e.g. serial0.txt

thanks @okurz  you pointed me to a right **file** where I can add a note regarding this patch. But I disagree about exact place which you mention. My point is that in current form this patch not just about uploading extra log file to a job. Currently it changing fully a way how we deciding about uploading **all** logs to every job so it would be confusing if we will put info about such generic functionality below some certain feature. Please check place which I selected - WDYT ?

File: docs/WritingTests.asciidoc (hunk 12 above)
Comment (asmorodskyi) [reply docs/WritingTests.asciidoc]: done

**Comment by asmorodskyi** [comment]
also @Martchus  please check if now it is ok

File: docs/WritingTests.asciidoc [hunk 13]
```diff
@@ -1408,3 +1408,13 @@ state of the specified git refspec. For example the asset caching
 functionality will override the +PRODUCTDIR+ variable. The above method can
 still be used in this case in the common case that the test schedule does not
 need to be changed or in case a custom schedule is defined by +SCHEDULE+.
+
+==Controling list of log files to upload
```
Comment (okurz) [inline docs/WritingTests.asciidoc]: space after `=`

File: docs/WritingTests.asciidoc [hunk 14]
```diff
@@ -1408,3 +1408,13 @@ state of the specified git refspec. For example the asset caching
...
+
+During every job run worker produces various log files.
```
Comment (okurz) [inline docs/WritingTests.asciidoc]: *the* worker

File: docs/WritingTests.asciidoc [hunk 15]
```diff
@@ -1408,3 +1408,13 @@ state of the specified git refspec. For example the asset caching
...
+By specifying +AUTOINST_UPLOAD_FROM_POOL+ variable with comma separated list of filenames
```
Comment (okurz) [inline docs/WritingTests.asciidoc]: `By specifying the variable +AUTOINST_UPLOAD_FROM_POOL+ with a comma separated list of filenames`

File: docs/WritingTests.asciidoc [hunk 16]
```diff
@@ -1408,3 +1408,13 @@ state of the specified git refspec. For example the asset caching
...
+it is possible to control which files should be uploaded to job results. Default set of files
```
Comment (okurz) [inline docs/WritingTests.asciidoc]: your line wrapping is off, please fix that

File: docs/WritingTests.asciidoc [hunk 17]
```diff
@@ -1408,3 +1408,13 @@ state of the specified git refspec. For example the asset caching
...
+is defined in https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Worker/Job.pm ( +UPLOAD_FROM_POOL+ constant ).
+Keep in mind that definition of +AUTOINST_UPLOAD_FROM_POOL+ variable will overwrite default list. So in case you
```
Comment (okurz) [inline docs/WritingTests.asciidoc]: I suggest to turn the whole section around:

By default openQA uploads a set of file as defined in … . You can override this by specifying +AUTOINST_UPLOAD_FROM_POOL+ …

File: lib/OpenQA/Worker/Job.pm [hunk 18]
```diff
@@ -36,6 +36,9 @@ has 'client';
 has 'developer_session_running';
 has 'upload_results_interval';
 
+#default list of files which would be uploaded from pool after job execution.
```
Comment (okurz) [inline lib/OpenQA/Worker/Job.pm]: * space after `#`
* s/would be/will be/

File: docs/WritingTests.asciidoc (hunk 13 above)
Comment (asmorodskyi) [reply docs/WritingTests.asciidoc]: done

File: docs/WritingTests.asciidoc (hunk 15 above)
Comment (asmorodskyi) [reply docs/WritingTests.asciidoc]: done

File: docs/WritingTests.asciidoc (hunk 16 above)
Comment (asmorodskyi) [reply docs/WritingTests.asciidoc]: not so easy to do when you not using awesome vim , but MS Visual Studio Code :D but I hope now all is fine

File: lib/OpenQA/Worker/Job.pm (hunk 18 above)
Comment (asmorodskyi) [reply lib/OpenQA/Worker/Job.pm]: done

File: docs/WritingTests.asciidoc (hunk 14 above)
Comment (asmorodskyi) [reply docs/WritingTests.asciidoc]: done

File: docs/WritingTests.asciidoc (hunk 17 above)
Comment (asmorodskyi) [reply docs/WritingTests.asciidoc]: For me it looks like I already have such structure just not exactly same wording : 
yours "By default openQA uploads a set of file as defined in "  is equal "Default set of files
is defined in " . "You can override this by specifying +AUTOINST_UPLOAD_FROM_POOL+" is equal to "definition of +AUTOINST_UPLOAD_FROM_POOL+ variable will overwrite default list ". And they going in same order as you asked . 

So I can not get what you actually mean when saying : 
> turn the whole section around 

?

File: docs/WritingTests.asciidoc [hunk 19]
```diff
@@ -1408,3 +1408,13 @@ state of the specified git refspec. For example the asset caching
...
+
+== Controling list of log files to upload
+
+During every job run the worker produces various log files. By specifying the variable +AUTOINST_UPLOAD_FROM_POOL+
+with a comma separated list of filenames it is possible to control which files should be uploaded to job results.
+Default set of files is defined in https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Worker/Job.pm
+( +UPLOAD_FROM_POOL+ constant ).
+Keep in mind that definition of +AUTOINST_UPLOAD_FROM_POOL+ variable will overwrite default list. So in case you want
+just add variable to list you need to specify your file plus all default. To see all files available to upload examine
+/var/lib/openqa/pool/<worker_id> during test execution.
```
Comment (okurz) [inline docs/WritingTests.asciidoc]: How about you sprinkle in some articles? ;)

Comment (Martchus) [reply docs/WritingTests.asciidoc]: Or add a hint to read the this paragraph with Russian accent :-)

Comment (asmorodskyi) [reply docs/WritingTests.asciidoc]: @Martchus "Spasiba" comrad :P

Comment (okurz) [reply docs/WritingTests.asciidoc]: @Martchus good idea. I already read out the paragraph to @asmorodskyi with Russian accent to prove my point ;)

Comment (asmorodskyi) [reply docs/WritingTests.asciidoc]: please check latest version ...

**Comment by asmorodskyi** [comment]
> I don't like this idea. Asking users to be aware of any future development of openQA is like a big middle finger to the user.
> 
> I kind of miss the underlying user story here.

Initially I just wanted to be able to get `virtio_console1.log` for **some** jobs . So I needed so way that openQA uploaded `virtio_console1.log` only on special request . The reason for that because mainly this log contain full content of other files so might be big and mainly useless , but in cases any issues with offline upload we need a way to get it for investigation.  
Initially I create  a patch which just append  `virtio_console1.log` to existing list of files to upload. Later during discussion @okurz suggested that this list should not be responsibility of openQA to "know" this list so we came up with current idea. 
Will you like this idea if I will revert patch to state when I was appending needed log file to hard-coded list which stored here ?

**Comment by Martchus** [comment]
Maybe you can change it so `AUTOINST_UPLOAD_FROM_POOL` would not replace but extend the existing list of files?

**Comment by cfconrad** [comment]
Fine for me as well.

**Comment by coolo** [comment]
If virtio_console1.log is created by openQA/isotovideo I'm all for uploading it unconditionally.

**Comment by coolo** [comment]
If you want it less hard coded create an index file in isotovideo that openQA reads and uploads referenced files. But don't leave it to the tester to know what files might be interesting

**Comment by asmorodskyi** [comment]
> If you want it less hard coded create an index file in isotovideo that openQA reads and uploads referenced files. But don't leave it to the tester to know what files might be interesting

not sure why you saying that I force tester to know list of files to upload. After merge of this patch average _tester_ free to use openQA as he/she did it before and don't care about list of files to upload. But for quite a long  many people keep telling that main users of openQA is actually  _openQA developers_ and this kind of users should know what files are uploaded to openQA so they will be capable to fulfill  this  variable is appropriate  list. Fallback to _append_ instead of _overwrite_ behavior will cause loose flexibility to remove some files from upload

**Comment by asmorodskyi** [comment]
> If virtio_console1.log is created by openQA/isotovideo I'm all for uploading it unconditionally.

so something like #2291 would be ok ?

**Comment by coolo** [comment]
> But for quite a long many people keep telling that main users of openQA is actually _openQA developers_ 

Not sure who that many people are and how you define openQA developer, but I count about a dozen developers and about a hundered test contributors.

File: docs/WritingTests.asciidoc (hunk 17 above)
Comment (okurz) [reply docs/WritingTests.asciidoc]: Your current text reads like this: "You can specify files. But by default you do not need to. However you can" :)

So I suggest to "turn it around" by starting with the simple case when one does not do anything, the default case. Then afterwards explain that this can be overriden. By this you also do not need to repeat again what is the order of precedence.

```suggestion
- By specifying +AUTOINST_UPLOAD_FROM_POOL+ variable with comma separated list of filenames
- it is possible to control which files should be uploaded to job results. Default set of files
- is defined in https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Worker/Job.pm ( +UPLOAD_FROM_POOL+ constant ).
- Keep in mind that definition of +AUTOINST_UPLOAD_FROM_POOL+ variable will overwrite default list.
+ By default openQA uploads a set of file as defined in the constant +UPLOAD_FROM_POOL+
+ in https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Worker/Job.pm .
+ You can override this set in the test variable +AUTOINST_UPLOAD_FROM_POOL+.
```

**Comment by asmorodskyi** [comment]
@okurz - updated

**Comment by okurz** [comment]
need to resolve conflicts

**Comment by asmorodskyi** [comment]
done

File: docs/WritingTests.asciidoc [hunk 20]
```diff
@@ -1408,3 +1408,9 @@ state of the specified git refspec. For example the asset caching
 functionality will override the +PRODUCTDIR+ variable. The above method can
 still be used in this case in the common case that the test schedule does not
 need to be changed or in case a custom schedule is defined by +SCHEDULE+.
+
+== Controling list of log files to upload
```
Comment (okurz) [inline docs/WritingTests.asciidoc]: s/Controling/Controlling/ and *the* list

Comment (asmorodskyi) [reply docs/WritingTests.asciidoc]: done

**Comment by asmorodskyi** [comment]
> It's ok for me but please attend coolo's comment [#2269 (review)](https://github.com/os-autoinst/openQA/pull/2269#pullrequestreview-280644557)
Because : 
1. I already got what I wanted with https://github.com/os-autoinst/openQA/pull/2291
2. don't want to invest more time into convincing  someone why this patch make sense 

I am closing this PR. Feel free to cherry-pick changes and try to convince @coolo that it is right thing to do
